### ISSUE TITLE
Add additional install options for Linux

### ISSUE THREAD

**Issue by chartgerink** [hypergraph-xyz/desktop#401]
Based on @wolass's comment in https://github.com/hypergraph-xyz/desktop/issues/392#issuecomment-793449301, opening this issue to provide a place for people to upvote the provision of additional install options for Linux, beyond the current `.AppImage`.

It is possible to build [`.rpm`, `.deb` files](https://www.electron.build/configuration/linux) using the electron builder.

**Comment by juliangruber** [comment]
snap is also an option, among others. See https://www.electron.build/configuration/linux#LinuxConfiguration-target

**Comment by chartgerink** [comment]
@wolass there can be differences of opinion, but please do not categorize different perspectives as inferior. That's your moral judgment that you bring, closes down the discussion, and is not in line with our code of conduct.

**Comment by wolass** [comment]
Well, just to clarify - the term "inferior" used by me in the post before
should not be considered as ethical judgement. It does not at all
classifies the use of any of the installation options for unix system on an
ethical plane.

The inferiority to which I was referring to is regarding the **security**
of the system for the user. Certainly having to maintain just one package
manager is superior than having to maintain multiple sources.

Also please note that the appImage, flatpacks and snaps all contain
dependencies in their files that might already be satisfied on an installed
unix system therefore leading to redundancies, and this is definitely a
problem.

And on top of that I do not discourage further discussion. And if you find
that Flatpacks / snaps / appImage are necessary for the users that is fine
with me (because every unix user is their own system's administrator and
has to know what they are doing). But I would appreciate if you would also
consider more reliable methods of installation that do not share the
shortcomings of the above forms of installation i.e., redundant package
install and lack of central update options through central package managers
(which are in my personal opinion making Snaps/Flatpacks and AppImage
installs inferior to apt / pacman installs).

I hope my wording is now clarified and the discussion can proceed.



Wojciech Francuzik <http://fracoscientific.com/>


wt., 9 mar 2021 o 12:47 Chris Hartgerink <notifications@github.com>
napisał(a):

**Comment by juliangruber** [comment]
Are you proposing to publish all node modules this app uses as individual debs?

**Comment by wolass** [comment]
@juliangruber - definitely not. 

I am proposing to provide hypergraph as at least  `.deb` and `pacman`  linux targets. 
I don't know what you mean by "node modules" but i suspect this has to do with how electron apps are built. 

1. If I understand the link you provided above (https://www.electron.build/configuration/linux#LinuxConfiguration-target) correctly (I have null experience in electron) it should be possible to provide at least the standard dependencies the app uses in the `depends` argument. Therefore it would greatly decrease the 'bloat'.

2. If there are dependencies that can not be satisfied this way (and for the app itself!) - I propose to make a simple ppa repository that can be added as source to the apt. Therefore the app will be automatically updated if there would be any security updates necessary for the WHOLE system, just by running `apt update && apt upgrade` . And I mean the whole system... really. 

No fiddling with appImage hubs,  flatpack hubs or snappy snaps (god... this name send a chill down my spine).

here is another article why these things actually are destroying linux experience for user like me. 
https://hackaday.com/2020/06/24/whats-the-deal-with-snap-packages/

I moved to linux to have granular control over my system. 
With the emergence of these formats this control is getting lost. 

Anyway, if you want me to provide more evidence why hypergraph should be provided as .deb / pacman packages (ppas) let me know. I dont want to start a rant here.

**Comment by juliangruber** [comment]
Hypergraph uses electron and Node.js packages as dependencies. I doubt publishing Node.js packages as linux packages makes sense. With electron, that's an interesting perspective, and I'm not sure if that can be done. Apps often rely on a specific version of electron, and chances of you using two apps that want the same electron version are very low. So actually, the single package format isn't introducing bloat, for most cases - as far as I understand it.

In terms of packages as .deb / pacman etc for ease of installation, I'm for it, as long as it doesn't mean too much development overhead. Bear in mind that developers will have to maintain VMs for all of the supported packaging version operating systems, in order to properly test them.

However, as far as I understand it, there's not much to do about the bloat argument, electron isn't yet set up to be distributed like this. And also, I also 100% understand where you're coming from, and I would like electron apps to be distributed efficiently to gnu/linux as well. I just don't think it's possible yet to the extend that you're asking for.

**Comment by wolass** [comment]
@juliangruber thank you for considering providing alternative (better, non-inferior, superior) install options for linux users. 

Now, I have to address a burning matter and I will do it here as it refers to my posts above. 

TLDR;
Today I have recived an email from @chartgerink informing me that I have broken the code of conduct by using the word "inferior" to describe AppImage packages. 
He wants me to apologize for this here. I will not do such a thing and I resign from this whole comunity. I no longer want to be a contributor to Hypergraph and I want my account on Hypergraph closed. But before I go this is my final post. 

 
Today I recived this email: 

```
Dear Wojciech Francuzik,

The liberate science code of conduct committee received a report regarding your behavior in our community. After considering this matter, the committee regards your behavior as a violation of our code of conduct.

Specifically, it regards the categorization of different perspectives as inferior in this GitHub discussion. It was highlighted as shutting down discussion, hence, not in line with the CoC, yet you continued the use of this terminology, without acknowledging its impact.

No action is taken against you at this moment beyond noting this incident for future reference. You are allowed to continue participating in our community in good faith.

In case you'd like to, we would appreciate and take note of a comment in the GitHub discussion where you indicate you chose your words poorly and will be more considerate in how your words impact others in the future.

In case you are interested in understanding the impact of this, you can find more information below.
"Inferior" is a word that has been historically used to legitimize various systems of oppression.

Deeming a different perspective in the community as inferior does not invite the option for discussion, and may prove off-putting or triggering for people who have been marginalized under these systems of oppression.

We will not correspond further about this incident report and consider this matter closed.

--
Kind regards,
Chris Hartgerink
Chair of liberate science code of conduct committee

```

I take this matter very seriously. I only took part in this conversation after Chris invited me to report issues with installing Hypergraph as I am a GNU Linux user. After I have presented my expertise above (stating that the AppImage software is more bloated option than the .deb packages, becase of redundant dependecies) he disliked the fact that I used the word "inferior". 

After I clarified that I do not mean anything offensive and that I am clearly referring to SOFTWARE not ideas or people, he just went on and decided to use the official organ of his company being the Code of Conduct Comitte to give me a reprimend. 

The email states that there has been a report filled and that "a committee" made a collaborative decision regarding my word choices.

First off, Chris, let me remind you how github works. In The File `CONTRIBUTING.md` you provide a link to a CoC, and this is the only valid CoC that applies. In your email you linked a document that is not a part of this repo. If I would agree to abide to a CoC it would only be the one which is in this repository. I suggest you sort out this problem for future cases, as there probably be plenty. The link from Notion does not have a version control so you can change it to your liking everytime you want to exclude someone from your community who is willing to exercise their right to free speach. So if you would truly abide to open scholarship principles you would stick to github CoC and have it versioned.  

Secondly, the CoC writes: (https://github.com/hypergraph-xyz/cli/blob/main/CODE_OF_CONDUCT.md)

```
Examples of unacceptable behavior by participants include:

- The use of sexualized language or imagery and unwelcome sexual attention or advances
- Trolling, insulting/derogatory comments, and personal or political attacks
- Public or private harassment
- Publishing others' private information, such as a physical or electronic address, without explicit permission
- Other conduct which could reasonably be considered inappropriate in a professional setting
```

My use of the word "inferior" does not meet any of the examples above. If you wish to exclude certain commonly used words from the dictionary so that the contributors of this repo do not use them - I strongly advise you to put this information into the CoC in form of a "blacklist". I hope this word is allowed. 

Next, there is also one huge problem with the CoC committee - namely, the fact that you were involved in having anything to say in my "case". Not only you were most probably the person filing the report, but you are at the same time the Chair of the CoC committee, and most importantly our relationship has a clear conflict of interest that you probably failed to disclose to other members of the committee. 

Namely, we used to be friends in Berlin and we met through our common friend at that time Jon Tennant, @protohedgehog. After Jon was accused by the OpenCon - It was you who was involved in  his removal from the Open Science Mooc - a project Jon funded. You were acting as a member of the CoC Committee at that time. A thing that you cerainly find plesure in, because it puts you in a position of power. 

Back then you also did not act based on facts but rather unbased accusations. Without Jon, this great project dissolved. **I have contacted you personally back then in 2019 to defend Jon and prevent you from banning him from praticipating in his own project. But you failed to reconsider. That led to a serious conflict of interest between us**, that in my humble opinion is the reason for you giving me an official reprimend. I will not have this. But take note - this kind of behaviour is not protecting the community - it is devastating to all of its members. 

I therefore command libscie to release all information you may have gathered regarding my person and release it to me including the CoC Committee "investigation". 

Although, I wanted to see hypergraph grow - because I belive that micropublications may change the face of scholarly publishing - I cannot be a member of a community where its leader fails put his personal grudge behind him, and reprimends members that have difference of opinions (not targeted at any person), and where a Code of Conduct committee is made of 3 people that do not question one another. Chris, if you can't see why this is a bad thing - then you need more external advisors. Someone who would actually NOT be afraid of loosing a job by not following your command. I advise you to revisit your CoC Committee and put more people on it. You, as the leader of this organisation CANNOT be a member of the CoC committee, because it leads to situations like this one. You simply can't be the police, prosecutor and the judge in one person if you want to have a healthy community.

Tu sum up:
1. Relese the CoC report on the above mentioned case from your email (You can blur out the name of the person who filled the report - because it is kind of clear who did it)
2. Confirm that you accept my resignation from libscie community and close my account on Hypergraph removing all my personal data. 
3. Addmit that you failed to disclose your conflict of interest with me regarding Jon Tennant's OpenCon Saga to other members of the CoC Committee.
4. Resign from being the Chair of the Code of Conduct Committee

Best regards

Dr. med. Wojciech Francuzik

**Comment by chartgerink** [comment]
I will close this thread now, as it is no longer about the original issue. 

If anybody would like to continue discussion on additional install options for Linux, I invite you to open up a new issue.

**Comment by johav** [comment]
Actually, I would like to see this issue being re-addressed and resolved professionally and openly and here, thank you. 
A concerned member of the Open Science community
